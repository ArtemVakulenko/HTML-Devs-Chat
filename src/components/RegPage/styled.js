import styled from 'styled-components';

export const StAuthContainer = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const StAuthView = styled.div`
  display: grid;
  grid-template-rows: repeat(4,1fr);
  row-gap: 15px;
  justify-items: center;
  align-items: center;
  background: ${({ colors, theme }) => colors[theme].secondary};
  border-radius: 10px;
  padding: 20px 60px;
`;
export const StErrorContainer = styled.div`
  min-height: 30px;
`;
