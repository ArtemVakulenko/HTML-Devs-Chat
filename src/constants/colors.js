export const colors = {
  dark: {
    main: '#0E1621',
    secondary: '#17212B',
    input: '#242F3D',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#fff',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#0E1621',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#488ccf',
  },
  light: {
    main: '#FFFFFF',
    secondary: '#ECECEC',
    input: '#D1D1D1',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#000',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#FFFFFF',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#2B5278',
  },
};
