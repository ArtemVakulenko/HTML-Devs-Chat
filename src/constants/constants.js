export const langs = {
  arabian: 'ar',
  russian: 'ru',
  english: 'en',
};

export const cookieAge = 20000;
