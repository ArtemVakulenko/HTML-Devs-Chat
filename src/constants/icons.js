export const icons = {
    changeThemeIcon: '../assets/img/changeThemeIcon.png',
    defaultUserIcon: '../assets/img/defaultUserIcon.png',
    dropDownIcon: '../assets/img/dropDownIcon.png',
    logoIcon: '../assets/img/logoIcon.png',
    logOutIcon: '../assets/img/logOutIcon.png',
    settingIcon: '../assets/img/settingsIcon.png',
    closeModalDark: '../assets/img/closeModalDark.png',
    sendMessage: '../assets/img/sendMessage.png',
    showPass: '../assets/img/view.png',
    hidePass: '../assets/img/hide.png',
    themeDark: '../assets/img/moon.png',
    themeLight: '../assets/img/sun.png',
};
