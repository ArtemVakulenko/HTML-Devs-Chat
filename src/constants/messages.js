export const MESSAGES_KEYS = {
  invalidLogin: 'invalidLogin',
  invalidPassword: 'invalidPassword',
  invalidConfirm: 'invalidConfirm',
  userIsReg: 'userIsReg',
  notExist: 'notExist',
  wrongPass: 'wrongPass',
  abstract: 'something went wrong',
  successReg: 'successReg',
  successAuth: 'successAuth',
  unauth: 'unAuth',
};
