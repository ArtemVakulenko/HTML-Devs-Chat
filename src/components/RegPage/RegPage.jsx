import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { MESSAGES_KEYS } from 'constants';
import { NotificationManager } from 'react-notifications';
import { useTranslation } from 'react-i18next';
import Title from '../_common/Title';
import Button from '../_common/Button';
import AuthInput from '../_common/AuthInput';
import CustomLink from '../_common/CustomLink';
import { StAuthContainer, StAuthView } from './styled';
import { useTheme } from '../hooks';

const RegPage = ({ message, signUpRequest }) => {
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  const redirect = useMemo(() => {
    if (message && message !== MESSAGES_KEYS.successReg) {
      NotificationManager.error(t(message), t('Attention'), 3000);
    }
    if (message === MESSAGES_KEYS.successReg) {
      return <Redirect to="/" />;
    }
    return null;
  }, [message]);
  const onKeyDown = (e) => {
    if (e.key === 'Enter') signUpRequest();
  };
  return (
    <StAuthContainer onKeyDown={onKeyDown}>
      <StAuthView colors={colors} theme={theme}>
        <Title title="registration" />
        <AuthInput type="login" page="auth" />
        <AuthInput type="password" page="auth" />
        <AuthInput type="confirm" page="auth" />
        <Button onClick={signUpRequest} value="registerBtn" />
        <CustomLink path="/" title="toAuth" />
      </StAuthView>
      {redirect}
    </StAuthContainer>
  );
};

export default RegPage;

RegPage.propTypes = {
  message: PropTypes.string.isRequired,
  signUpRequest: PropTypes.func.isRequired,
};
