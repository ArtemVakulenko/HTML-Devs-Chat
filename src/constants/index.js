import { colors } from './colors';
import { icons } from './icons';
import { MESSAGES_KEYS } from './messages';
import { url } from './url';
import { LOGIN, PASSWORD } from './regExp';
import { langs, cookieAge } from './constants';

export { url, colors, icons, MESSAGES_KEYS, LOGIN, PASSWORD, langs, cookieAge };
