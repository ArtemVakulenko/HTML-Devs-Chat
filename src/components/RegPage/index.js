import { connect } from 'react-redux';
import { signUpRequest } from 'store/auth/authActions';
import { getErrorMessage } from 'store/auth/selectors';
import RegPage from './RegPage';

const mapStateToProps = (state) => ({
  message: getErrorMessage(state),
});

const mapDispatchToProps = {
  signUpRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(RegPage);
