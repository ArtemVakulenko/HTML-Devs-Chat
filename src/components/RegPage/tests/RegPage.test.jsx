import configureStore from 'redux-mock-store';
import React from 'react';
import { NotificationManager } from 'react-notifications';
import { mountSmart, shallowSmart } from '../../../__tests__/testHelper';
import RegPage from '../RegPage';
import { MESSAGES_KEYS } from '../../../constants';

const mockStore = configureStore();
const store = mockStore({
  auth: {
    auth: {
      login: '',
      password: '',
      confirm: '',
    },
  },
});
jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('react-notifications', () => ({
  NotificationManager: {
    error: jest.fn(),
  },
}));
jest.mock('../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));

describe('Input', () => {
  const props = {
    message: 'MockValue',
    signUpRequest: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = shallowSmart(<RegPage {...props} />, store);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StAuthContainer', () => {
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('styled__StAuthContainer')).toHaveLength(1);
  });
  it('Should have StAuthView', () => {
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('styled__StAuthView')).toHaveLength(1);
  });
  it('Should have Title', () => {
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('Title')).toHaveLength(1);
  });
  it('Should have AuthInput', () => {
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('AuthInput')).toHaveLength(3);
  });
  it('Should have Button', () => {
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('Should have CustomLink', () => {
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('CustomLink')).toHaveLength(1);
  });
  it('Should have notification error', () => {
    mountSmart(<RegPage {...props} />, store);
    expect(NotificationManager.error).toHaveBeenCalled();
  });
  it('Should have notification error & no redirect', () => {
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('Redirect')).toHaveLength(0);
  });
  it('Should have success', () => {
    props.message = MESSAGES_KEYS.successReg;
    const component = mountSmart(<RegPage {...props} />, store);
    expect(component.find('Redirect')).toHaveLength(1);
  });
});
