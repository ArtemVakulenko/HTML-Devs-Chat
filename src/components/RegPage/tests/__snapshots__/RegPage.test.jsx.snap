// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match snapshot 1`] = `"<div class=\\"styled__StAuthContainer-gfSiSj iMPmgL\\"><div class=\\"styled__StAuthView-iNEnXV jqqTSt\\"><h1 class=\\"styled__AppTitle-bMNsHC jWDbjD\\"></h1><div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"text\\" value=\\"\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/></div><div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"password\\" value=\\"\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/><img src=\\"../assets/img/view.png\\" class=\\"styled__StHidePass-kZfxvt kfuTnG\\"/></div><div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"password\\" value=\\"\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/><img src=\\"../assets/img/view.png\\" class=\\"styled__StHidePass-kZfxvt kfuTnG\\"/></div><button class=\\"styled__StButton-iwFefC fohYxd\\"></button><a href=\\"/\\"></a></div></div>"`;
